### PR TITLE
Feature/get tag by name

### PR DESCRIPTION
**Pull request by FredoVelcro**
added Tag::getByName() method

### CODE REVIEW

File: models/DataObject/CalculatedFieldInterface.php [hunk 1]
```diff
@@ -0,0 +1,39 @@
...
+ *
+ * @copyright  Copyright (c) Pimcore GmbH (http://www.pimcore.org)
+ * @license    http://www.pimcore.org/license     GPLv3 and PEL
+ */
+
+namespace Pimcore\Model\DataObject;
+
+use Pimcore\Model\DataObject\Data\CalculatedValue;
+
+interface CalculatedFieldInterface
```
Comment (jdreesen) [inline models/DataObject/CalculatedFieldInterface.php:22]: This is unrelated, I think?

File: models/Element/Tag.php [hunk 2]
```diff
@@ -73,6 +73,24 @@ public static function getById($id)
...
+     *
+     * @return self|null
+     */
+    public static function getByName($name)
+    {
+        try {
+            $class = new self();
+            $id = $class->getDao()->getByName($name);
+            if ($id) {
+                return self::getById($id);
```
Comment (jdreesen) [inline models/Element/Tag.php:87]: Wouldn't this fetch the data twice?

File: models/Element/Tag.php [hunk 3]
```diff
@@ -73,6 +73,24 @@ public static function getById($id)
...
+    /**
+     * @param string $name
+     *
+     * @return self|null
+     */
+    public static function getByName($name)
+    {
+        try {
+            $class = new self();
+            $id = $class->getDao()->getByName($name);
```
Comment (jdreesen) [inline models/Element/Tag.php:85]: It doesn't seem like `Dao::getByName()` returns anything.

**Comment by brusch** [comment]
@FredoVelcro any update?

**Comment by FredoVelcro** [comment]
sorry didn't have time to recreate a proper PR without the unrelated code...

**Comment by brusch** [comment]
Closed due to inactivity. 
Just create a new PR as soon as you have time again 😊 Thanks!
